### ISSUE TITLE
Allow reuse of parts of the image block's functionality in other blocks

### ISSUE THREAD

**Issue by SeanDS** [WordPress/gutenberg#18548]
**Is your feature request related to a problem? Please describe.**
I'd like to have my custom block display an image and let the user resize it, add an alt description, crop it, etc. but not replace it using the media manager. To avoid implementing functionality that  would essentially be identical to that of the core image block, I'd like to just reuse parts of the core image block's functionality (e.g. resize/alt/crop/caption/etc.) in my own block without having to copy any of the code.

**Describe the solution you'd like**
Ideally I would be able to subclass the React component or embed parts of the core image block inside my own block, but turn off the functionality that lets the user replace the image with another from the media library. (Is subclassing possible? I don't think the React components are bundled with the compiled version of Gutenberg, and subclassing is generally advised against in React.)

As an aside, it would also surely benefit the Gutenberg project if some core "image-display" React component could be shared between the core image, gallery and media+text blocks. This would reduce duplicated code and allow this duplicated functionality (i.e. displaying and altering images) to use the same code base and benefit from bug fixes etc., and allow people like me to use the component in our own blocks.

**Describe alternatives you've considered**
I could copy the code from the existing image block, but this will quickly become out of date as the image block receives regular updates and bug fixes.

**Comment by paaljoachim** [comment]
Hey @SeanDS Sean

Thank you for creating this issue!
A lot of time has passed.
Is this issue still valid?

@ItsJonQ Jon. I am thinking you might want to know about this issue.

**Comment by SeanDS** [comment]
Yes, a lot of time has passed. I would still be interested in such a feature. The key aspect I'm interested in is the ability to reuse the image block (with all the tools for cropping, resizing, etc.) but switch off the user's ability to change the image `src` or choose another image from their media library.

I've not kept much of an eye on it, but it looks like the inner blocks feature might allow this - can you or anyone else confirm if what I wrote in the original description might work with inner blocks (or any other recent Gutenberg feature)?

**Comment by paaljoachim** [comment]
Hey Sean.
I do not know. Hopefully Jon can drop on by and add a comment.

**Comment by youknowriad** [comment]
Closing as addressed: the building blocks are exported for reuse now — MediaPlaceholder, MediaReplaceFlow, ResizableBox, and the media-utils upload pieces — and the InnerBlocks + template approach covers embedding a full Image block with locked-down UI. If a specific internal remains unreachable, a scoped export request would be easy to consider. Thanks!
